internal/sqlparser/parser.go

```diff
@@ -171,6 +175,10 @@ func ParseSQLMigration(r io.Reader, direction Direction, debug bool) (stmts []st
 			case "+goose NO TRANSACTION":
 				useTx = false
 				continue
+
+			case "+goose ENVSUB ON":
+				useEnvsub = true
+				continue
```

Comment:
@mfridman I believe ability to disable further substition in a file is important, there could be huge initial migrations where one could be interested in replacing environment variables only in certain small parts of it and ignore the rest.

```suggestion

			case "+goose ENVSUB ON":
				useEnvsub = true
				continue

			case "+goose ENVSUB OFF":
				useEnvsubst = false
				continue
```